src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,114 @@
+package org.sasanlabs.service.vulnerability.commandInjection;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.util.Strings;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.sasanlabs.service.exception.ServiceApplicationException;
+import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.Assert;
+
+/*
+ * @author Feng-Yuan Yang coolesaltw@hotmail.com
+ */
+public class CommandInjectionTest {
+
+    private CommandInjection commandInjection;
+
+    private String localhost = "localhost";
+    Supplier<Boolean> trueCondition = () -> true;
+    Supplier<Boolean> falseCondition = () -> false;
+    String successReturnStr = "PING SUCCESS";
+
+    @BeforeEach
+    void setUp() throws IOException {
+        commandInjection = Mockito.spy(new CommandInjection());
+
+        StringBuilder stringBuilder = new StringBuilder().append(successReturnStr);
+
+        // Real localhost ping is not able to operate in CI, Mock instead
+        doReturn(stringBuilder)
+                .when(commandInjection)
+                .getResponseFromPingCommand(eq(localhost), any(Supplier.class));
+
+        doReturn(trueCondition).when(commandInjection).getSupplier(true);
+
+        doReturn(falseCondition).when(commandInjection).getSupplier(false);
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        // Act
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(localhost);
+
+        // Assert
+        Assert.isTrue(
+                successReturnStr.equals(responseEntity.getBody().getContent()),
+                "must get response");
```

Comment:
message is not descriptive and can cause problems later. please add descriptive message.